### DISCUSSION TITLE
N4TDX.org dashboard config.js

### DISCUSSION THREAD

**Discussion by midair55** [VA3HDL/hamdashboard#21] [General]
// Includes the configuration for image rotation.   Cheers!  Steve - N5NG

const aURL = [
  ["add10d", "Back", "#", "1"],
  // ["ff9100", "Refresh", "#", "1"],
  ["0dd1a7", "Help", "#", "1"],
  ["2196F3", "Launches", "https://www.spacelaunchschedule.com/category/fl", "1"],
  [
        "2196F3",
        "Forecast",
        "https://forecast.weather.gov/MapClick.php?lat=28.3937444&lon=-80.6064444",
        "1.0",
  ],
  [
        "2196F3",
        "Wx Outlook",
        "https://forecast.weather.gov/wwamap/wwatxtget.php?cwa=MLB&wwa=hazardous%20weather%20outlook",
        "2.0",
  ],
  [
        "FF000F",
        "Wx Alerts",
        "https://alerts.weather.gov/search?zone=FLC009", // ,FLC095,FLC117",
        "1"
        // ,FLC009"; - Brevard
        // ,FLC095"; - Orange
        // ,FLC117"; - Seminole
  ],
  [
        "2196F3",
        "Wx Obs",
        // "https://www.weather.gov/mlb/obs",
        "https://forecast.weather.gov/data/obhistory/KXMR.html",
        // "https://www.weather.gov/wrh/timeseries?site=KXMR",
        "1.0",
  ],
  [
    "2196F3",
    "Hi-Res",
    //"https://radar.weather.gov/?settings=v1_eyJhZ2VuZGEiOnsiaWQiOiJ3ZWF0aGVyIiwiY2VudGVyIjpbLTgxLjc3MSwyOC40MDVdLCJsb2NhdGlvbiI6Wy04MC43NzksMjguNDA2XSwiem9vbSI6OC41MTAxMjg5NjY2MzMxNTJ9LCJhbmltYXRpbmciOnRydWUsImJhc2UiOiJzdGFuZGFyZCIsImFydGNjIjpmYWxzZSwiY291bnR5IjpmYWxzZSwiY3dhIjpmYWxzZSwicmZjIjpmYWxzZSwic3RhdGUiOmZhbHNlLCJtZW51Ijp0cnVlLCJzaG9ydEZ1c2VkT25seSI6ZmFsc2UsIm9wYWNpdHkiOnsiYWxlcnRzIjowLjgsImxvY2FsIjowLjYsImxvY2FsU3RhdGlvbnMiOjAuOCwibmF0aW9uYWwiOjAuNn19",
    "https://moe.met.fsu.edu/~mnissenbaum/RadarArchive/KMLB/loop.html",
    "1",
  ],
  [
    "2196F3",
    "Lightning",
    "https://map.blitzortung.org/#7.05/28.33/-81.38",
    "1",
  ],
  [
    "2196F3",
    "Wind",
    // "https://earth.nullschool.net/#current/wind/surface/level/orthographic=-80.60,28.39,10000/loc=-80.593,28.330",
    "https://www.ventusky.com/?p=28.25;-80.85;8&l=wind-10m",
    "1",
  ],
  [
    "2196F3",
    "Rain Fall",
    "https://maps.cocorahs.org/?maptype=precip&datetype=daily&center=28.3893,-80.6088",
    //"https://www.cocorahs.org/Maps/ViewMap.aspx?state=fl&county=bv",
    // "https://www.cocorahs.org/Maps/GetMap.aspx?state=FL&county=BV&type=precip",
    "1",
  ],
  [
    "2196F3",
    "AQI",
    "https://ca.dep.state.fl.us/mapdirect/?map=9e94c40966164a31b162557510e7f0cc",
    "1",
  ],
  [ "9C33FF",
    "Aircraft",
    "https://globe.adsbexchange.com/?lat=28.378&lon=-80.709&enableLabels&extendedLabels=1&zoom=11",
    "1",
  ],
  [ "9C33FF",
    "Marine",
    "https://www.marinetraffic.org/ships-in-port/Port%20Canaveral/us/type-Port",
    "1",
  ],
  //[ "33E3FF",
   // "Pier Cam",
    //"https://www.youtube.com/embed/F8y1uqHJsLc?autoplay=1",
    //"1",
  //],
  [ "33E3FF",
    "Beach Cam",
    // "https://www.surfguru.com/widget/surf-forecast/cocoa-beach-pier-north?aud=0&rwcl=11354",
    "https://www.surfguru.com/widget/surf-forecast/cocoa-beach-pier-north?chk=78&aud=1&rwcl=11356", // South of Pier
    "1",
  ],
  [ "33E3FF",
    "Port Cam",
    "https://www.youtube.com/embed/FTw5xuq2swo?si=jNiQRYDfl37RElao?autoplay=1",
    "1",
  ],
  [ "33E3FF",
    "Cape Cam",
    "https://www.youtube.com/embed/Jm8wRjD3xVA?autoplay=1",
    "1",
  ],
// Right side slideout
  //["2196F3", "TIME.IS", "https://time.is", "1", "R"],
  ["2196F3", "DX Cluster", "https://dxcluster.ha8tks.hu/map/", "1", "R",],
  [ "2196F3",
    "Contest",
    "https://www.contestcalendar.com/fivewkcal.html",
    "1",
    "R",
  ],
  [ "2196F3",
    "POTA",
    "https://pota.app",
    "1",
    "R",
  ],
  [
    "2196F3",
    "Aurora",
    "https://services.swpc.noaa.gov/images/animations/ovation/north/latest.jpg",
    "1",
    "R",
  ],
  [
    "2196F3",
    "NOAA D-RAP",
    "https://services.swpc.noaa.gov/images/animations/d-rap/global/d-rap/latest.png",
    "1.4",
    "R",
  ],
  [
    "2196F3",
    "MUF",
    "https://prop.kc2g.com/renders/current/mufd-normal-now.svg",
    "1.0",
    "R",
  ],
  [
    "2196F3",
    "WinLink",
    "https://cms.winlink.org:444/maps/propagation.aspx",
    "1.0",
    "R",
  ],
  [
    "2196F3",
    "10M Prop",
    "https://www.tvcomm.co.uk/g7izu/Autosave/NA_ES_AutoSave.JPG",
    "1.0",
    "R",
  ],
  [
    "2196F3",
    "APRS",
    "https://aprs.to/?center=28.3867,-80.7344&zoom=11",
    "1.0",
    "R",
  ],
  [
    "2196F3",
    "ISS Loc",
    "https://www.heavens-above.com/orbitdisplay.aspx?icon=iss&width=600&height=300&mode=M&satid=25544",
    "2.0",
    "R",
  ],
  [
    "2196F3",
    "ISS Passes",
    "https://www.heavens-above.com/PassSummary.aspx?satid=25544&lat=28.3893&lng=-80.6088&loc=Cape_Canaveral&alt=0&tz=UCT",
    "1.0",
    "R",
  ],
  // [
    // "2196F3",
    // "Grey Line",
    // "https://www.timeanddate.com/scripts/sunmap.php?iso=now",
    // "https://www.timeanddate.com/scripts/sunmap.php?obj=moon&iso=now",
    // "1.2",
    // "R",
  // ],
  [
    "2196F3",
    "Grey Line",
    "https://www.timeanddate.com/scripts/sunmap.php?obj=sun&iso=now",
    "1",
    "R",
  ],
  [
    "2196F3",
    "Band Plan",
    "https://kd4atw.org/wp-content/uploads/2012/05/band_plan.png",
    "1",
    "R",
  ],
];

// Dashboard items
// Structure is Title, Image Source URL
// [Title, Image Source URL],
// the comma at the end is important!
// You can't add more items because there are only 12 placeholders on the dashboard
// but you can replace the titles and the images with anything you want.
const aIMG = [
  [
    "Tropical Outlook",
    "https://www.nhc.noaa.gov/xgtwo/two_atl_0d0.png",
    "https://www.nhc.noaa.gov/xgtwo/two_atl_2d0.png",
    "https://www.nhc.noaa.gov/xgtwo/two_atl_7d0.png",
    "https://www.cpc.ncep.noaa.gov/products/precip/CWlink/ghaz/gth_full.png",
  ],
  [
    "Tropics",
        "https://cdn.star.nesdis.noaa.gov/GOES16/ABI/SECTOR/taw/Sandwich/GOES16-TAW-Sandwich-900x540.gif",
        "https://cdn.star.nesdis.noaa.gov/GOES16/ABI/SECTOR/car/Sandwich/GOES16-CAR-Sandwich-1000x1000.gif",
        "https://cdn.star.nesdis.noaa.gov/GOES16/ABI/SECTOR/se/Sandwich/GOES16-SE-Sandwich-600x600.gif",
        "https://cdn.star.nesdis.noaa.gov/GOES16/ABI/SECTOR/gm/Sandwich/GOES16-GM-Sandwich-1000x1000.gif",
        "https://cdn.star.nesdis.noaa.gov/GOES16/ABI/SECTOR/eus/Sandwich/GOES16-EUS-Sandwich-1000x1000.gif",
  ],
  ["Forecast & Activity",
        "https://www.wpc.ncep.noaa.gov/noaa/noaa.gif",
        "https://www.wpc.ncep.noaa.gov/noaa/noaad2.gif",
        "https://www.wpc.ncep.noaa.gov/noaa/noaad3.gif",
        "https://www.spc.noaa.gov/products/exper/day4-8/day48prob.gif",
        "https://www.wpc.ncep.noaa.gov/threats/final/hazards_d3_7_contours.png",
        "https://www.wpc.ncep.noaa.gov/qpf/fill_94qwbg.gif",
        "https://www.spc.noaa.gov/products/outlook/day3otlk_0730.gif",
        "https://www.spc.noaa.gov/products/exper/enhtstm/imgs/enh_2000.gif",
        "https://www.spc.noaa.gov/exper/mesoanalysis/activity_loop.gif",
        "https://www.weather.gov/wwamap/png/US.png",
        "https://www.spc.noaa.gov/products/watch/validww.png",
        "https://www.spc.noaa.gov/climo/reports/today.gif",
        "https://data.mesonet.org/data/public/noaa/metar/maps/realtime/latest.tair.png",
        // "https://s.w-x.co/staticmaps/acttemp_1280x720.jpg?crop=16:9",
        // "https://www.usairnet.com/weather/images/current-temperature.png",
        // "https://origin.wpc.ncep.noaa.gov/basicwx/allfcsts_loop_ndfd.gif",
  ],
  [
    "Satellite",
    "https://cdn.star.nesdis.noaa.gov/GOES16/GLM/SECTOR/se/EXTENT3/GOES16-SE-EXTENT3-600x600.gif",
    "https://cdn.star.nesdis.noaa.gov/GOES16/GLM/SECTOR/eus/EXTENT3/GOES16-EUS-EXTENT3-1000x1000.gif",
    "https://cdn.star.nesdis.noaa.gov/GOES16/GLM/SECTOR/gm/EXTENT3/GOES16-GM-EXTENT3-1000x1000.gif",
    "https://cdn.star.nesdis.noaa.gov/GOES16/GLM/CONUS/EXTENT3/GOES16-CONUS-EXTENT3-625x375.gif",
    // "https://cdn.star.nesdis.noaa.gov/GOES16/ABI/CONUS/Sandwich/20242020836-20242021251-GOES16-ABI-CONUS-Sandwich-625x375.gif",
  ],
  [
    "RADAR",
    "https://radar.weather.gov/ridge/standard/KMLB_loop.gif",
    // "https://radar.weather.gov/ridge/standard/SOUTHEAST_loop.gif",
    "https://radar.weather.gov/ridge/standard/SOUTHEAST_loop.gif",
    "https://radar.weather.gov/ridge/standard/CONUS-LARGE_loop.gif",
    // "https://radar.weather.gov/ridge/standard/CONUS_loop.gif",
  ],
  [
    "Lightning",
    "https://images.lightningmaps.org/blitzortung/america/index.php?animation=florida",
    "https://images.lightningmaps.org/blitzortung/america/index.php?animation=usa",
  ],
  [
    "Forecast Today",
    "https://www.weather.gov/images/mlb/graphicast/image_full1.png",
    "https://www.weather.gov/images/mlb/graphicast/image_full2.png",
    "https://www.weather.gov/images/mlb/adas/temp_sfc_loop.gif",
    "https://www.weather.gov/images/mlb/adas/heatindex_sfc_loop.gif",
    "https://www.cpc.ncep.noaa.gov/products/stratosphere/uv_index/uvi_map.gif",
  ],
  [
    "Tonight & Tomorrow",
    "https://www.weather.gov/images/mlb/graphicast/image_full3.png",
    "https://www.weather.gov/images/mlb/graphicast/image_full4.png",
  ],
  [
    "2 Day Hourly Forecast",
    "https://forecast.weather.gov/meteograms/Plotter.php?lat=28.3893&lon=-80.6088&wfo=MLB&zcode=FLZ447&gset=18&gdiff=3&unit=0&tinfo=EY6&ahour=0&pcmd=1111111111&lg=en&indu=1!1!1!&dd=3&bw=0&hrspan=48",
  ],
  [
        "", // Risks
        "https://www.weather.gov/images/mlb/ghwo/SevereThunderstormsDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/ThunderstormWindDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/TornadoDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/LightningDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/ExcessiveRainfallDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/ExcessiveHeatDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/WindDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/FogDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/HighSurfDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/RipRiskDay1.jpg",
        "https://www.weather.gov/images/mlb/ghwo/MarineHazardDay1.jpg",
        // "https://www.weather.gov/images/mlb/GHWO/skywarn.png",
  ],
  [
    "Tide Table & Wave Height",
    "https://wave.marineweather.net/itide/tides/png/fl_trident_pier_port_canaveral.png",
    "https://www.ndbc.noaa.gov/plot_met.php?station=41113&meas=wvht&uom=E",
    "https://www.ndbc.noaa.gov/plot_met.php?station=41009&meas=wvht&uom=E",
    "https://media.raven.news/images/wx/bn9/60min/wave_heights_atl.jpg",
  ],
  [
        "", // Traffic
        "https://fl511.com/map/Cctv/1138--11",
        "https://fl511.com/map/Cctv/1140--11",
        "https://fl511.com/map/Cctv/1142--11",
        "https://fl511.com/map/Cctv/1143--11",
  ],
  ];

// Define tile image refresh/rotate rate in milliseconds.
// Use zero (0) for a single static image.
// Caution: Short intervals should be avoided to reduce the possibility the
// image server detects a Denial of Service attack and blocks the request.

const tileDelay = [
        // Top row
        600000,
        120000,
        20000,
        120000,
        // Middle row
        60000,
        60000,
        300000,
        300000,
        // Bottom row
        1800000,
        10000,
        900000,
        60000,
];

**Comment by midair55** [comment]
The N4TDX config.js is a copy of N5NG's dashboard with the exception of call sign.  The config.js is available as a .txt file at: https://qsl.net/n5ng/config.txt

**Reply by VA3HDL** [reply]
How do you get QSL.net to update your dashboard when you make changes?
I recently started testing this site as an alternative option to host the dashboard and every time I make changes I cannot see these reflected on the dashboard.
Any thoughts? 
Thanks

**Reply by VA3HDL** [reply]
Mine is at https://www.qsl.net/va3hdl/dashboard/

**Comment by midair55** [comment]
I've noticed that as well.  The 'update' is not seen after uploading a new config.js and refreshing the browser.   Same after clearing the browser's cache; no change.   Same no-change seen using a different browser...   Magically the update is seen after a period of time.  I have not determined how long it takes before the update is 'seen'.   Just a guess....  I wonder if there is some type of periodic virus scan on the QSL server and the updated config.js isn't used until scanned and passes....???

I've also noticed that the URL changes at times with an "n" before the call sign --  Example:   first go to https://qls.net/n5ng   then add "HAM" to the end of the URL.  This results in: https://qsl.net/n/n5ng//HAM/

________________________________

**Reply by VA3HDL** [reply]
You right. I just waited and now I can see the updated config applied to the dashboard.

n or the v in my case is the first letter of the callsign and is probably a way to keep the directories organized on the server, regardless your site works fine with this URL https://qsl.net/n5ng/HAM/
so you can remove the "n/" part from your bookmarks

**Comment by midair55** [comment]
Correct.  I don’t include the “n” in the URL but found it odd when it
appeared.
